blocklib/timing/src/event_definitions.hpp

```diff
@@ -0,0 +1,428 @@
+#ifndef GR_DIGITIZERS_EVENT_DEFINITIONS_HPP
+#define GR_DIGITIZERS_EVENT_DEFINITIONS_HPP
+
+static const std::array<ImColor, 13> bpcidColors{
+        // colors taken from: https://git.acc.gsi.de/fcc-applications/common-context-widget-reactor/src/branch/master/common-context-widget-fx/src/main/resources/de/gsi/fcc/applications/common/contextwidget/fx/StylesColorsBase.css
+        ImColor{0x00, 0x70, 0xC0}, // 0x0070C0 - materials
+        ImColor{0x00, 0xB0, 0x50}, // 0x00B050 - bio
+        ImColor{0x5C, 0xB7, 0xC3}, // 0x5CB7C3 - cbm
+        ImColor{0x70, 0x30, 0xA0}, // 0x7030A0 - nustar-basic
+        ImColor{0xC6, 0x59, 0x11}, // 0xC65911 - machine
+        ImColor{0xFF, 0xDA, 0x4A}, // 0xFFDA4A - nustar-r3b
+        ImColor{0xC4, 0x8B, 0xD0}, // 0xC48BD0 - nustar-she-c
+        ImColor{0xE1, 0x4C, 0xFF}, // 0xE14CFF - nustar-she-p
+        ImColor{0xFF, 0xFF, 0x00}, // 0xFFFF00 - nustar-hades
+        ImColor{0xFF, 0xA6, 0x32}, // 0xFFA632 - plasma
+        ImColor{0x00, 0xFC, 0xD6}, // 0x00FCD6 - appa-pp
+        ImColor{0x80, 0x80, 0x80}, // 0x808080 - undefined
+        ImColor{0x9e, 0x9e, 0x9e}  // 0x9e9e9e - invalid
+};
+
+// generic colormap for timing ids
+static std::array<ImColor, 20> colorlist { // gemerated with http://medialab.github.io/iwanthue/
+        ImColor{78,172,215,120}, ImColor{200,76,41,120}, ImColor{85,199,102,120}, ImColor{186,84,191,120},
+        ImColor{101,173,51,120}, ImColor{117,98,204,120}, ImColor{169,180,56,120}, ImColor{211,76,146,120},
+        ImColor{71,136,57,120}, ImColor{209,69,88,120}, ImColor{86,192,158,120}, ImColor{217,132,45,120},
+        ImColor{102,125,198,120}, ImColor{201,160,65,120}, ImColor{199,135,198,120}, ImColor{156,176,104,120},
+        ImColor{168,85,112,120}, ImColor{55,132,95,120}, ImColor{203,126,93,120}, ImColor{118,109,41,120},
+};
+
+// https://www-acc.gsi.de/wiki/Timing/TimingSystemGroupsAndMachines#Groups_for_Operation
+static const std::map<uint16_t, std::pair<std::string, std::string>> timingGroupTable{
```

Comment:
Note that this table and as well `eventNrTable` might change over time. Afaik they are generated out of some controls DB. Might be no good idea to hardcode them.

Here a script which creates C++ header files and XML files out of the DB. You might want to use that for the cmake codegen in order to always work with an up-to-date event/group-list:  

https://git.acc.gsi.de/FE-tools/DB/src/branch/master/wr-timing-configuration/createTimingConfiguration.py

Walled garden GSI/ACO repo .... here the script, if you dont have access: [createTimingConfiguration.py.zip](https://github.com/fair-acc/gr-digitizers/files/13485455/createTimingConfiguration.py.zip)

The generated .h / xml files are located here: `asl754:/common/usr/timing/groupsAndEvents/`

Reply:
Thanks for the hints, I don't want the build to depend on being inside the GSI network, but long term they are definitely helpful to keep these definitions up to date.

Reply:
Yea, would be good to be able to build it everywhere.

Just an idea: How about some optional build flag `GENERATE_TIMING_CONFIG_FROM_DB` ? If unset, some fallback table could be used.